ui/src/variables/selectors/index.tsx

```diff
@@ -32,6 +32,48 @@ export const getVariablesForOrg = (
   return getVariablesForOrgMemoized(state.variables.variables, orgID)
 }
 
+export const getVariablesForDashboard = (
+  state: AppState,
+  dashboardID: string
+): Variable[] => {
+  const {
+    variables: {variables, values},
+  } = state
+
+  let variablesForDash = []
+
+  const variablesIDs = Object.keys(get(values, `${dashboardID}.values`))
+
+  variablesIDs.forEach(variableID => {
```

Comment:
I think this might supposed to be:

```
  variablesIDs.forEach(variableID => {
    const variable = get(variables, `${variableID}.variable`)

    if (variable) {
      variablesForDash.push(variable)
    }
  })
```